tests/integration/serverless/serverless.yml

```diff
@@ -188,9 +188,9 @@ functions:
           consumer:
             Fn::GetAtt:
               - KinesisStreamConsumer
-              - ConsumerARN
+              - StreamARN
           maximumRecordAgeInSeconds: 120
-          startingPosition: LATEST
+          startingPosition: TRIM_HORIZON
```

Comment:
I switched to `TRIM_HORIZON` here because it's more reliable. My understanding of the `serverless` tests is that infra gets stood up in a fairy async way. Using `LATEST`, if we send records to Kinesis before the event source mapping is declared and becomes fully active then those records will never be processed. `TRIM_HORIZON` ensures that all records will be processed regardless of the order in which those operations occur.
I agree that it would be great to test both configurations. Ideally we could do this using the `get_event_source_mapping()` API by polling the state of the mapping and waiting until it becomes `ACTIVE` before we send records. Unfortunately our implementation of this API is not yet robust enough to do this. It always reports a mapping as ACTIVE even if the associated EventSourceListener instance hasn't finished coming up. In order to make this work we will need to refactor the EventSourceListener class to make use of some kind of EventSourceListenerManager singleton that can accurately understand and manage the state of the various event source mappings (or at least that's the right way to do it IMO).